### PR TITLE
Remove cards optimistically

### PR DESCRIPTION
**Pull request by DylanDylann**
### Explanation of Change
Remove cards optimistically in both cardList and cards_{workspaceAccountID}_ExpensifyCard when users deactivate cards on all places: Wallet, Member Detail Page, Expensify Card

### Fixed Issues

$ https://github.com/Expensify/App/issues/51738
https://github.com/Expensify/App/issues/51739

### Tests

QA Steps
- [x] Verify that no errors appear in the JS console

### QA Steps

Section 1:
1. Go to workspace settings > Members.
2. Click on your own profile.
3. Click Add card.
4. Assign a virtual card to yourself.
5. Go to Wallet.
6. Click on the card.
7. Click Report virtual card fraud.
8. Click Deactivate card.
9. Go back to members and click on your own profile.
10. Note that the deactivated card is removed.
11. Go to Wallet.
12. Note that the deactivated card is removed.

Section 2: 
1. Go to workspace settings > Expensify Card.
2. Click on the card > Deactivate card.
3. Deactivate the card.
4. Go to Members.
5. Click on the member whose card is deactivated.
6. Note that the deactivated card is removed

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/16de2e0e-47cd-4e0e-9dfa-36746b1a8e9b

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/76fb58a0-c272-4dff-8e53-8a15827b45fa

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/f3110a08-77a2-418f-9634-a6fbaee55be6

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/8a593a7b-a9d5-45fa-9909-f41f3d691a74

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/670eda7a-5575-437f-be4c-77eb0246f802

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/018c5afe-7b9b-42e1-acd1-8e0d40d4006a

</details>

### CODE REVIEW

**Comment by jayeshmangwani** [comment]
@DylanDylann Pressing 'Deactivate Card' on the 'Report Virtual Card Fraud' page briefly shows a 'Not Found' page.

https://github.com/user-attachments/assets/6736fccc-babc-46be-98d2-5329c8dab438

**Comment by jayeshmangwani** [comment]
Also, please upload the videos to the checklist

**Comment by DylanDylann** [comment]
@jayeshmangwani I see

> @DylanDylann Pressing 'Deactivate Card' on the 'Report Virtual Card Fraud' page briefly shows a 'Not Found' page.

What do you think if we close RHP after clicking deactivate card

**Comment by DylanDylann** [comment]
> Also, please upload the videos to the checklist

Upload later. I hope it doesn't block your reviewing

**Comment by jayeshmangwani** [comment]
> What do you think if we close RHP after clicking deactivate card

@DylanDylann I didn't get that. Can you elaborate on what you're trying to say here?

**Comment by jayeshmangwani** [comment]
> Upload later. I hope it doesn't block your reviewing

No, it doesn’t. This is just a reminder to follow the checklist

**Comment by DylanDylann** [comment]
> @DylanDylann I didn't get that. Can you elaborate on what you're trying to say here?

@jayeshmangwani 

RCA: The card is removed optimistically --> Not found Page


So I suggest we close Report virtual card fraud Page, when clicking on deactivate card

**Comment by DylanDylann** [comment]
Or could you have any suggestion for this problem?

**Comment by jayeshmangwani** [comment]
@DylanDylann Yes we can close the RHP, we are doing same under the Member ->  Cards -> Deactivate Card

**Comment by jayeshmangwani** [comment]
@mountiny What are your thoughts on this? We are getting a '[Not Found](https://github.com/Expensify/App/pull/51841#issuecomment-2451389911)' page for the 'Deactivate Card' under 'Report Fraud.' [Should we close the RHP](https://github.com/Expensify/App/pull/51841#issuecomment-2451447529) on the 'Deactivate Card' button?

**Comment by jayeshmangwani** [comment]
@DylanDylann This issue isn’t solved entirely; we can still see the 'Not Found' page when back navigation occurs

https://github.com/user-attachments/assets/ef4719b8-fddb-41ef-a33c-7a699ab62e82

**Comment by DylanDylann** [comment]
@jayeshmangwani Could you please check again?

**Comment by jayeshmangwani** [comment]
Code changes look good. I am testing a few things and found something strange happening while loading the wallet. I'm not sure if it’s related to this PR, but I will add the reproduction steps and the issue in a moment.

**Comment by DylanDylann** [comment]
All yours

**Comment by jayeshmangwani** [comment]
Hey @DylanDylann , can you please check if this works fine for me, or if something is messed up on my account? I am getting constant loading for the "Report Virtual Card Fraud" page's button.


https://github.com/user-attachments/assets/9f4f4832-55df-4525-8e77-f45e043cd0d5

**Comment by DylanDylann** [comment]
> Hey @DylanDylann , can you please check if this works fine for you, or if something is messed up on my account? I am getting constant loading for the "Report Virtual Card Fraud" page's button.

@jayeshmangwani Why it is loading? I don't get this problem

**Comment by DylanDylann** [comment]
> Thanks for making the changes @DylanDylann . I just have a question, if you have context on it: why are we not showing a confirmation modal on Wallet -> Report Fraud -> Deactivate before deactivating the card? We are showing it on the other two pages, so I just wanted to know if you have any context on this.

It maybe our implementation before. Not sure

**Comment by jayeshmangwani** [comment]
>  I'm not sure if it’s related to this PR, but I will add the reproduction steps and the issue in a moment.


Finally, I was able to reproduce this on a new account.

Repro Steps:

1. Create a card and assign it to yourself.
2. Go to Wallet -> Assigned Cards -> Press on the card -> Report Virtual Card Fraud -> Deactivate Card.
3. You will be navigated to Wallet and receive an error for that card indicating that something went wrong while deactivating it.
4. Refresh the page, and the card disappears.

So, if we encounter an error, why does the card disappear? This may not be related to this PR, but I found it while testing, so I'm pasting it here for visibility.

**Comment by jayeshmangwani** [comment]
> Why it is loading? I don't get this problem

@DylanDylann Please ignore this, as I am unable to reproduce it on another account. If I find the exact repro steps, I will post them; otherwise, we can ignore it for this PR.

**Comment by DylanDylann** [comment]
@jayeshmangwani BE removed data after reloading page, right?

**Comment by jayeshmangwani** [comment]
@DylanDylann But if we encounter an error when deactivating, then why is the backend removing the data that I wondering?

**Comment by jayeshmangwani** [comment]
As I said before, this is not from this PR, but I noticed it and wanted to mention it here.

**Comment by DylanDylann** [comment]
@jayeshmangwani Is this ready for checklist?>

**Comment by jayeshmangwani** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/50145dba-13c4-49fd-8931-cfade5b2569b




</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/a64cdd12-ddd0-4290-a7c5-4f4a1229b975




</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/1c5a5534-b596-4fda-b73a-37a5d43a3e4e




</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/d9d115da-f78d-4a40-80bb-15ffb7fcc60d




</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/cdd721d3-34a5-4fa3-9f65-b30e87cf9407


https://github.com/user-attachments/assets/c75421ec-9fc8-43b1-8671-96165a4ff597




</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/3310ff91-bd6e-422b-9e75-e3e647dffb53




</details>

**Review by jayeshmangwani — APPROVED** [review]

File: src/libs/actions/Card.ts [hunk 1]
```diff
@@ -37,7 +37,8 @@ type IssueNewCardFlowData = {
     data?: Partial<IssueNewCardData>;
 };
 
-function reportVirtualExpensifyCardFraud(cardID: number) {
+function reportVirtualExpensifyCardFraud(card?: Card) {
+    const cardID = card?.cardID ?? -1;
```
Comment (mountiny) [inline src/libs/actions/Card.ts:41]: I m still not sure what is the best way to handle these. I feel like that if the cardID is not available we should cut this earlier, but not sure about the best practice here

File: src/libs/actions/Card.ts [hunk 2]
```diff
@@ -46,6 +47,20 @@ function reportVirtualExpensifyCardFraud(cardID: number) {
...
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.CARD_LIST,
+            value: {
+                [cardID]: null,
+            },
+        },
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.WORKSPACE_CARDS_LIST}${card?.fundID ?? '-1'}_${CONST.EXPENSIFY_CARD.BANK}`,
```
Comment (mountiny) [inline src/libs/actions/Card.ts:59]: same here, if the fundID is not available, this is kinda useless

File: src/pages/settings/Wallet/ReportVirtualCardFraudPage.tsx [hunk 3]
```diff
@@ -36,6 +36,13 @@ function ReportVirtualCardFraudPage({
 
     const prevIsLoading = usePrevious(formData?.isLoading);
 
+    const submit = useCallback(() => {
+        Navigation.dismissModal();
+        InteractionManager.runAfterInteractions(() => {
+            Card.reportVirtualExpensifyCardFraud(virtualCard);
+        });
+    }, [virtualCard]);
```
Comment (mountiny) [inline src/pages/settings/Wallet/ReportVirtualCardFraudPage.tsx:44]: Why did we need this callback here and not in some other pages?

**Review by mountiny — APPROVED** [review]
Changes look good to me

**Comment by mountiny** [comment]
Thank you for thorough testing @jayeshmangwani

**Pull request merged**